### DISCUSSION TITLE
Количество треков в файле SpotifyRecentTracks

### DISCUSSION THREAD

**Discussion by Bondian** [Chimildic/goofy#197] [Помощь с ошибкой] [answered]
Строки из файла конфигурации:

> Отслеживание истории включено
UserProperties.setProperty('ON_SPOTIFY_RECENT_TRACKS', 'true');

> Количество треков: 80000
UserProperties.setProperty('COUNT_RECENT_TRACKS', '80000');

Но в файле только 50000 треков:

```
function testRecentTracks() {

  let cachedRecentTracks = Cache.read('SpotifyRecentTracks.json');
  console.log('Количество треков SpotifyRecentTracks', cachedRecentTracks.length);

  // на всякий случай
  let recentTracks = RecentTracks.get();
  console.log('Количество треков recentTracks', recentTracks.length);
}
```
Журнал выполнения
13:38:12	Примечание	Выполнение начато
13:43:30	Информация	Количество треков SpotifyRecentTracks 50000
13:43:30	Информация	Количество треков recentTracks 50000
13:38:18	Примечание	Выполнение завершено

Зашёл на гугл диск. Последнее изменение файла SpotifyRecentTracks.json - свежее - за последний час.
В чём ещё может быть подвох?

**Comment by Chimildic** [comment, marked as answer]
Проверьте какое количество сохранено, а не просто указано.
```js
function checkProperty(){
  console.log(UserProperties.getProperties().COUNT_RECENT_TRACKS);
}
```

**Reply by Bondian** [reply]
> Проверьте какое количество сохранено, а не просто указано.
> 
> ```js
> function checkProperty(){
>   console.log(UserProperties.getProperties().COUNT_RECENT_TRACKS);
> }
> ```

Журнал выполнения
15:26:52	Примечание	Выполнение начато
15:32:07	Информация	50000
15:26:55	Примечание	Выполнение завершено

**Reply by Bondian** [reply]
Всё, понял. Запустил конфиг. Не знал, что его надо запускать после изменения. Думал, что он связан с основным файлом библиотеки, и больше не требует никаких действий, кроме сохранения. Теперь всё в порядке

Журнал выполнения
15:29:40	Примечание	Выполнение начато
15:34:55	Информация	80000
15:29:44	Примечание	Выполнение завершено

**Reply by Chimildic** [reply]
То что пишем мы это просто код. Значения нужно сохранять в условный файл настроек, чтобы остальной код мог его читать и подставлять значения.
Я сейчас даже не вспомню почему отказался от принудительного пересохранения при каждом запуске. Может время выполнения съедало?..
